iroh/src/node.rs

```diff
@@ -653,6 +665,32 @@ impl<D: BaoMap + BaoReadonlyDb, C: CollectionParser> RpcHandler<D, C> {
         })
     }
 
+    fn list_incomplete_blobs(
+        self,
+        _msg: ListIncompleteBlobsRequest,
+    ) -> impl Stream<Item = ListIncompleteBlobsResponse> + Send + 'static {
+        let db = self.inner.db.clone();
+        let local = self.inner.rt.local_pool().clone();
+        futures::stream::iter(db.partial_blobs()).filter_map(move |hash| {
```

Comment:
can `partial_blobs()` block? (or rather it's iterator impl? if so wouldn't it be better to run the whole iterator on a `spawn_blocking` and send the results via a channel?

Reply:
The current idea is that fns like get, get_partial, and these iterators don't block.

So while the data can be on disk, the information about what you have must be in memory. If you want to have the ability to perform IO you would probably turn partial_blobs into a stream or a future that returns a stream.

Reply:
In general, things like spawn_blocking where they are needed are left to the store impl. We don't want to call spawn_blocking just in case and then slow down an in mem impl.

Reply:
Okay, so the requirement here is on the store implementation to make sure this doesn't have any bad blocking effects. Maybe add this to the docs of the trait, so it is not lost?

Reply:
For Map::get I would argue that this is already the case. But I will make it clear also for the iterators.